### PR TITLE
Python 3 for `initdb`

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Makes the necessary changes for `initdb.py` to work with Python 3.

### Testing

```
$ python initdb.py
```

Addresses https://issues.redhat.com/browse/PROJQUAY-164
Commits taken from https://github.com/quay/quay/pull/183

### CODE REVIEW

**Comment by alecmerdler** [comment]
I'm not sure if it's a problem that `test/data/test.db` was changed as a result. Should it be the same under Python 3?

**Comment by kurtismullins** [comment]
@alecmerdler Can you run `black` on this branch? It's currently failing in the "lint" stage.

File: util/security/secret.py [hunk 1]
```diff
@@ -25,5 +25,10 @@ def convert_secret_key(config_secret_key):
         secret_key = str(bytearray(list(map(ord, config_secret_key))))
 
     # Otherwise, use the bytes directly.
-    assert len(secret_key)
-    return "".join(itertools.islice(itertools.cycle(secret_key), 32))
+    assert len(secret_key) > 0
+    print(secret_key)
```
Comment (kurtismullins) [inline util/security/secret.py]: Not sure if this was intentional but we should probably pull this `print()` statement out. I assume it was there for debugging purposes.

File: util/security/secret.py [hunk 2]
```diff
@@ -25,5 +25,10 @@ def convert_secret_key(config_secret_key):
...
+
+    # FIXME(alecmerdler): Testing
+    secret_key = b'\xa3l\x9d}%\xa90\xf4\xa5\x86=/\x8d\xc4\n\x83'
```
Comment (kurtismullins) [inline util/security/secret.py]: I assume this was also left in for debugging purposes.

**Comment by kleesc** [comment]
You'll probably need to remove the enclosing `str()` in L13 and L25 in `util/security/secret.py`

File: util/security/secret.py [hunk 3]
```diff
@@ -22,8 +22,9 @@ def convert_secret_key(config_secret_key):
             pass
 
     if secret_key is None:
-        secret_key = str(bytearray(list(map(ord, config_secret_key))))
+        secret_key = bytearray(list(map(ord, config_secret_key)))
```
Comment (alecmerdler) [inline util/security/secret.py]: This is causing differences in `test/data/test.db` when running in Python 2 versus Python 3. Investigating.

**Comment by alecmerdler** [comment]
Closed by https://github.com/quay/quay/pull/183

**Pull request closed without merging**